webapp/src/main/java/com/box/l10n/mojito/service/thirdparty/ThirdPartyTMSSmartling.java

```diff
@@ -913,6 +1161,44 @@ private Stream<List<TextUnitDTO>> partitionPlurals(
     return partitionedStream(parameters, searchFunction);
   }
 
+  private Stream<List<TextUnitDTO>> partitionPlurals(
+      Long repositoryId,
+      String localeTag,
+      String skipTextUnitsWithPattern,
+      String skipAssetsWithPathPattern,
+      Set<String> pluralFixForLocales,
+      String includeTextUnitsWithPattern,
+      StatusFilter statusFilter) {
+
+    Function<TextUnitSearcherParameters, List<TextUnitDTO>> searchFunction =
+        textUnitSearcher::search;
+
+    if (pluralFixForLocales.contains(localeTag)) {
+      searchFunction =
+          searchFunction.andThen(
+              textUnits ->
+                  textUnits.stream()
+                      .filter(tu -> !MANY.toString().equalsIgnoreCase(tu.getPluralForm()))
+                      .collect(Collectors.toList()));
+    }
+
+    TextUnitSearcherParameters parameters =
+        baseParams(
+            repositoryId,
+            localeTag,
+            skipTextUnitsWithPattern,
+            skipAssetsWithPathPattern,
+            false,
+            false,
+            "%",
+            includeTextUnitsWithPattern,
+            statusFilter);
+
+    parameters.setOrderByTextUnitID(true);
+    parameters.setStatusFilter(statusFilter);
```

Comment:
For partitionSingulars & partitionPlurals -given the only difference between the new versions and the old ones the statusFilter, could we have just one implementation - where the other function calls it with e.g.: statusFilter = null? 

And then in the actual implementation we can do:
```
if (statusFilter != null) {  
        parameters.setStatusFilter(statusFilter);
}  
```

Reply:
(we could even call the single implementation with statusFilter=null in the calling code directly to avoid having thin wrapper if desired)